src/network/networkd-ndisc.c

```diff
@@ -199,6 +206,9 @@ static int ndisc_request_route(Route *in, Link *link, sd_ndisc_router *rt) {
         if (route->mtu == 0)
                 route->mtu = mtu;
 
+        if (route->hop_limit == 0 && hop_limit > 0)
```

Comment:
The second condition (`hop_limit > 0`) is redundant and not necessary.

Reply:
Updated thanks for reviewing !